executor-core/src/main/java/net/spals/appbuilder/executor/core/DefaultExecutorServiceFactory.java

```diff
@@ -105,24 +100,46 @@ public ScheduledExecutorService createScheduledThreadPool(
         return scheduledExecutorService;
     }
 
-    ExecutorService decorateExecutorService(final ExecutorService delegate) {
+    @Override
+    public Optional<Boolean> stop(final Key key) {
+        Objects.requireNonNull(key);
+        final Optional<ExecutorService> executoreService = Optional.ofNullable(executorServices.get(key));
+        if (executoreService.isPresent()) {
+            if (executoreService.get().isShutdown()) {
+                return Optional.of(false);
+            }
+            executoreService.ifPresent(x -> stopExecutorService(key, x));
+            return executoreService.map(x -> true);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Boolean> isTerminated(final Key key) {
+        Objects.requireNonNull(key);
```

Comment:
You've used the `@Nonnull` annotation in other places so why the change?

Reply:
I forgot. Good catch.